framework/src/bundle/BundleArchive.cpp

```diff
@@ -147,16 +148,16 @@ std::shared_ptr<BundleResourceContainer> BundleArchive::GetResourceContainer()
   return resourceContainer;
 }
 
-void BundleArchive::RegisterOpenedResource() const 
-{ 
-  std::lock_guard<std::mutex> lock(numOpenResourcesLock);
-  numOpenResources++; 
+void BundleArchive::RegisterOpenedResource() const
+{
+  uint32_t expected = 0;
+  while(!numOpenResources.compare_exchange_weak(expected, expected + 1));
 }
 
-void BundleArchive::UnregisterOpenedResource() const 
-{ 
-  std::lock_guard<std::mutex> lock(numOpenResourcesLock);
-  numOpenResources--;
+void BundleArchive::UnregisterOpenedResource() const
+{
+  uint32_t expected = 0;
+  while(!numOpenResources.compare_exchange_weak(expected, expected - 1));
   if (numOpenResources == 0) {
     resourceContainer->CloseContainer();
```

Comment:
unfortunately there's still a race between the time in which ```numOpenResources``` is atomically set and when it is evaluated by the if statement.